src/core/src/op/constant.cpp

```diff
@@ -61,524 +84,227 @@ ov::op::v0::Constant::Constant(const std::shared_ptr<ngraph::runtime::Tensor>& t
-            rc.push_back(std::to_string(value));
-        }
-        break;
-    case element::Type_t::u16:
-        for (uint32_t value : get_vector<uint16_t>()) {
-            rc.push_back(std::to_string(value));
-        }
-        break;
-    case element::Type_t::u32:
-        for (uint32_t value : get_vector<uint32_t>()) {
-            rc.push_back(std::to_string(value));
-        }
-        break;
-    case element::Type_t::u64:
-        for (uint64_t value : get_vector<uint64_t>()) {
-            rc.push_back(std::to_string(value));
+    }
+
+    template <ov::element::Type_t ET,
+              class T = fundamental_type_for<ET>,
+              typename std::enable_if<std::is_integral<T>::value && !std::is_same<T, int8_t>::value>::type* = nullptr>
+    static result_type visit(const Constant* const c, std::vector<std::string>& strs) {
+        for (auto&& v : c->get_vector<T>()) {
+            strs.push_back(std::to_string(v));
         }
-        break;
-    case element::Type_t::undefined:
-    case element::Type_t::dynamic:
-        OPENVINO_THROW("unsupported type");
     }
-#if defined(__GNUC__) && !(__GNUC__ == 4 && __GNUC_MINOR__ == 8)
-#    pragma GCC diagnostic pop
-#endif
 
-    return rc;
+    template <ov::element::Type_t ET,
+              typename std::enable_if<std::is_same<fundamental_type_for<ET>, int8_t>::value>::type* = nullptr>
+    static result_type visit(const Constant* const c, std::vector<std::string>& strs) {
+        for (auto&& v : c->cast_vector<int8_t>()) {
+            strs.push_back(std::to_string(v));
+        }
+    }
+};
+
+std::vector<std::string> Constant::get_value_strings() const {
+    std::vector<std::string> out;
+    using namespace ov::element;
+    IfTypeOf<boolean, bf16, f16, f32, f64, i4, i8, i16, i32, i64, u1, u4, u8, u16, u32, u64, nf4>::apply<
+        ValuesToString>(get_element_type(), this, out);
+    return out;
 }
 
-ov::Shape ov::op::v0::Constant::get_shape_val() const {
+Shape Constant::get_shape_val() const {
     OPENVINO_ASSERT(m_element_type.is_integral_number());
-    std::vector<int64_t> out_shape = cast_vector<int64_t>();
-    ov::Shape output_shape(shape_size(m_shape));
-    std::transform(out_shape.begin(), out_shape.end(), output_shape.begin(), [&](const int64_t& v) {
-        return (v > 0) ? v : 0;
-    });
-    return output_shape;
+    return convert_values_to<Shape>(cast_vector<int64_t>(), m_shape);
 }
 
-ov::Strides ov::op::v0::Constant::get_strides_val() const {
+Strides Constant::get_strides_val() const {
     OPENVINO_ASSERT(m_element_type == element::i64);
-    std::vector<int64_t> out_strides = cast_vector<int64_t>();
-    Strides output_strides(shape_size(m_shape));
-    std::transform(out_strides.begin(), out_strides.end(), output_strides.begin(), [&](const int64_t& v) {
-        return (v > 0) ? v : 0;
-    });
-    return output_strides;
+    return convert_values_to<Strides>(get_vector<int64_t>(), m_shape);
```

Comment:
looks like `cast_vector` should be used instead of `get_vector`

Reply:
The `cast_vector` checks if Constant value is in range of casted type, as `assert` checks that element type is `i64` there is no value conversion and additional validation is not required, `get_vector` can be used here.